### PR TITLE
TOC fixed in pages

### LINKED ISSUE #388: TOC missing on some pages
## Describe the bug

Table of contents is missing on the following pages: Binary Numbers, Digital Sequential circuits, Race around condition, 
## To Reproduce

Can be done via adding {:toc} on top of the page

## Expected behavior
TOC must be present on these pages

## Additional context
I am working on this issue

### PR DESCRIPTION
**Pull request by abhinavsri360**
A meaningful title for PR, not like: made changes to xyz.md

Fixes #
TOC has been fixed in the following pages: binary numbers, digital sequential circuits, and race around condition which was not present before.Closes #388 
Ref #

#### Changes done:
Added TOC

#### Screenshots
image
image
image

#### Preview Link(s): 
https://deploy-preview-389--tender-ardinghelli-ebb79f.netlify.app/docs/binary

#### ✅️ By submitting this PR, I have verified the following

- [x] Checked to see if a similar PR has already been opened 🤔️
- [x] Reviewed the contributing guidelines 🔍️
- [x] Sample preview link added (add the link(s) for all the pages changed/updated from the checks tab after checks complete)
- [x] Tried Squashing the commits into one

### CODE REVIEW

File: docs/binary.md [hunk 1]
```diff
@@ -24,7 +29,9 @@ Binary Number :: 11001
 
 ---
 
-### How do we count using binary?
+## Binary Counting
+
+<h2>How do we count using binary?</h2>
```
Comment (YashKumarVerma) [inline docs/binary.md:34]: Any particular reasons for using a heading tags in markdown?

File: docs/binary.md [hunk 2]
```diff
@@ -36,7 +43,7 @@ It is just like counting in decimal except we reach 10 much sooner.
 | ???          | But then there is no symbol for 2 ... what to do?  |
 
 
-#### Well how do we count in decimal?
+<h2>Well how do we count in decimal?</h2>
```
Comment (YashKumarVerma) [inline docs/binary.md:46]: Same.

File: docs/binary.md [hunk 3]
```diff
@@ -57,15 +64,15 @@ It is just like counting in decimal except we reach 10 much sooner.
...
+<h2>What happens in decimal?</h2>
 
 
 | Decimal       | Explanation            |
 |:-------------:|:----------------------:|
 |	99	 |	When you run out of digits, ...|
 |	100	 |	... start from back at **0** again, but carry **1** on the left|
 
-#### And that is what is done in binary ...
+<h2>And that is what is done in binary ...</h2>
```
Comment (YashKumarVerma) [inline docs/binary.md:75]: one of many. please check entire file.

**Comment by abhinavsri360** [comment]
The reason why I have replaced heading tags with the existing hashtags is that when I use the former in that case the particular text is included in TOC which is very long and looks ugly in UI, this tag will prevent that and not break the UI on the other side. There are 4 such tags as far as I remember, that was done purposely to avoid break-in TOC

**Comment by YashKumarVerma** [comment]
@Shivansh2407 @Shivansh2407 @schezfaz What do you think ?

**Comment by schezfaz** [comment]
I feel we should preserve the #-format for declaring headings since it is afterall a markdown file. 

To address the issues faced while using the # way of declaring headings, we can simply use the .no_toc tag so that it will be excluded from the TOC entirely.

File: docs/binary.md [hunk 4]
```diff
@@ -5,7 +5,12 @@ nav_order: 2
 ---
 
 # Binary numbers
-{: .no_toc }
```
Comment (schezfaz) [inline docs/binary.md:8]: Also, since the name of the page is _Binary Numbers_ itself, I don't think that we should be including it in the TOC. Becomes a little redundant. Especially since the corresponding index number for this heading in the TOC is _1_, making all other headings sub-headings to this page. Ideally, the subheadings, say _Introduction_, _Binary Counting_ etc. should be numbered as _1_, _2_ and so on respectively. 

TLDR; the **_.no_toc_** should be present in my opinion.

File: docs/jk_race_around_condition.md [hunk 5]
```diff
@@ -6,6 +6,12 @@ nav_order: 14
 
 # Race around condition
```
Comment (schezfaz) [inline docs/jk_race_around_condition.md:7]: This heading should not be present in the TOC. Same goes for all other files that have the page title included in the heading.

File: docs/binary.md (hunk 1 above)
Comment (Dishebh) [reply docs/binary.md:34]: Also, please follow the sentence case guidelines in headings mentioned [here](https://developers.google.com/style/capitalization)

**Comment by abhinavsri360** [comment]
Now this PR can be reviewed, thanks for suggesting changes @YashKumarVerma @schezfaz @Dishebh 
Let me know if any more are required, here's the preview link for the changes: https://deploy-preview-389--tender-ardinghelli-ebb79f.netlify.app/docs/binary.html

File: docs/binary.md (hunk 1 above)
Comment (abhinavsri360) [reply docs/binary.md]: Thanks @Dishebh

File: docs/binary.md (hunk 4 above)
Comment (abhinavsri360) [reply docs/binary.md:8]: updated

File: docs/binary.md [hunk 6]
```diff
@@ -52,12 +47,8 @@ It is just like counting in decimal except we reach 10 much sooner.
 | ???          | But then there is no symbol for 2 ... what to do?  |
 
 
-<<<<<<< Updated upstream
-<h2>Well how do we count in decimal?</h2>
-=======
 ## Well how do we count in Decimal?
```
Comment (schezfaz) [inline docs/binary.md:50]: Another small attention to detail, it is in best practices to have a smaller dimension heading under a given heading instead of having the heading and sub-heading as the same dimension. For example, < h2 > level headings should not have sub-headings of < h2 > level, instead they should be a dimension (or more) smaller, like < h3 > or < h4 >

Over here, _How do we count using binary?_  is a subheading to _Binary Counting_ and thus should not be at par with it's parent-heading level. I would suggest keep the subheadings at < h3 > i.e. `###` level and the headings at < h2 > i.e. `##` level

**Comment by schezfaz** [comment]
Looks good! Just a few more rectifications regarding the heading - subheading declarations. What do you feel?

File: docs/binary.md [hunk 7]
```diff
@@ -36,11 +32,10 @@ Binary Number :: 11001
 ---
 
 ## Binary Counting
-<<<<<<< Updated upstream
 
-<h2>How do we count using binary?</h2>
-=======
->>>>>>> Stashed changes
+## How do we count using binary?
```
Comment (schezfaz) [inline docs/binary.md:36]: Another small attention to detail, it is in best practices to have a smaller dimension heading under a given heading instead of having the heading and sub-heading as the same dimension. For example, < h2 > level headings should not have sub-headings of < h2 > level, instead they should be a dimension (or more) smaller, like < h3 > or < h4 >

Over here, _How do we count using binary?_ is a subheading to _Binary Counting_ and thus should not be at par with it's parent-heading level. I would suggest keep the subheadings at < h3 > i.e. ### level and the headings at < h2 > i.e. ## level

**Comment by abhinavsri360** [comment]
waiting for review

**Comment by abhinavsri360** [comment]
@YashKumarVerma

File: docs/binary.md [hunk 8]
```diff
@@ -104,11 +119,9 @@ Further, the respective binary digit is multiplied with the value present on top
 In this Case ::
 1x16 + 1x8 + 0x4 + 0x2 + 1x1 = 25 which is the decimal equivalent of the binary number 11001
 ```
-<p>Click on the '0' to change it to '1' and vice-versa</p>
 
-{% include binary.html %}
+## Signed and Unsigned numbers
```
Comment (Shivansh2407) [inline docs/binary.md:123]: Please make use of Sentence case

File: docs/binary.md [hunk 9]
```diff
@@ -80,9 +95,9 @@ It is just like counting in decimal except we reach 10 much sooner.
 | 111        | |
 | **1**000   |Start back at 0 again (for all 3 digits), add 1 on the left|
 
-## See how it is done in this little demonstration
+## Binary to Decimal Demonstration
```
Comment (Shivansh2407) [inline docs/binary.md:98]: Sentence case

**Comment by abhinavsri360** [comment]
> Please do the minor fixes @abhinavsri360 . Also can't we use any other alternative rather than using Html tags. Please let me know your views 👍

I have removed the HTML tags in the latest commits and they are now handled with hashtags

File: docs/binary.md (hunk 8 above)
Comment (abhinavsri360) [reply docs/binary.md:123]: Sure will fix these in a minute or so

File: docs/binary.md (hunk 9 above)
Comment (abhinavsri360) [reply docs/binary.md]: fixed @Shivansh2407

**Comment by abhinavsri360** [comment]
@Shivansh2407 can this be merged now?

**Comment by abhinavsri360** [comment]
@Shivansh2407 I think we should add a different tab/button to access pop_quiz should be present on the right to TOC such that it is notable when anyone accesses the page.
As far as adding Pop_quiz to TOC is concerned I would like to fix that in a different PR. As of now this PR closes #388 and let this be specific. 
Please let me know if any more changes are required.

**Comment by abhinavsri360** [comment]
Here's the preview link: https://deploy-preview-389--tender-ardinghelli-ebb79f.netlify.app/docs/binary
